Algorithm.Framework/Risk/CompositeRiskManagementModel.cs

```diff
@@ -0,0 +1,111 @@
+    /// </summary>
+    public class CompositeRiskManagementModel : RiskManagementModel
+    {
+        private readonly IRiskManagementModel[] _riskManagementModels;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CompositeRiskManagementModel"/> class
+        /// </summary>
+        /// <param name="riskManagementModels">The individual risk management models defining this composite model</param>
+        public CompositeRiskManagementModel(params IRiskManagementModel[] riskManagementModels)
+        {
+            if (riskManagementModels.IsNullOrEmpty())
+            {
+                throw new ArgumentException("Must specify at least 1 risk management model for the CompositeRiskManagementModel");
+            }
+
+            _riskManagementModels = riskManagementModels;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CompositeRiskManagementModel"/> class
+        /// </summary>
+        /// <param name="riskManagementModels">The individual risk management models defining this composite model</param>
+        public CompositeRiskManagementModel(params PyObject[] riskManagementModels)
+        {
+            if (riskManagementModels.IsNullOrEmpty())
+            {
+                throw new ArgumentException("Must specify at least 1 risk management model for the CompositeRiskManagementModel");
+            }
+
+            _riskManagementModels = new IRiskManagementModel[riskManagementModels.Length];
+
+            for (var i = 0; i < riskManagementModels.Length; i++)
+            {
+                if (!riskManagementModels[i].TryConvert(out _riskManagementModels[i]))
+                {
+                    _riskManagementModels[i] = new RiskManagementModelPythonWrapper(riskManagementModels[i]);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CompositeRiskManagementModel"/> class
+        /// </summary>
+        /// <param name="riskManagementModel">The individual risk management model defining this composite model</param>
+        public CompositeRiskManagementModel(PyObject riskManagementModel)
+            : this(new[] { riskManagementModel} )
+        {
+
+        }
+
+        /// <summary>
+        /// Manages the algorithm's risk at each time step. 
+        /// This method patches this call through the each of the wrapped models.
+        /// </summary>
+        /// <param name="algorithm">The algorithm instance</param>
+        /// <param name="targets">The current portfolio targets to be assessed for risk</param>
+        /// <returns>The new portfolio targets</returns>
+        public override IEnumerable<IPortfolioTarget> ManageRisk(QCAlgorithmFramework algorithm, IPortfolioTarget[] targets)
+        {
+            foreach (var model in _riskManagementModels)
+            {
+                targets = model.ManageRisk(algorithm, targets).ToArray();
```

Comment:
Interesting means of composing these -- simply put, the targets from portfolio construction get piped into risk model 1, then the risk-adjusted targets from risk model 1 get piped into risk model 2, 3, etc...

I think this is the correct way of doing this, nice work.

Reply:
Thanks @mchandschuh